### PR TITLE
Add bind defined values to docs

### LINKED ISSUE #1446: Bind documentation update
Add description of MATCH_MASK, BIND_USE_ATTR, BIND_EXEC_LOG, etc, to bind internals doc.

### PR DESCRIPTION
**Pull request by vanosg**
Found by: Geo
Patch by: Geo
Fixes: #1446 

One-line summary:
Add explanations of #define'd values commonly used in creating Tcl binds when building new modules

@thommey please review and fix anything- some of this is just guesswork from looking at code. Specifically, MATCH_MODE,  BIND_WANTRET, BIND_ALTER_ARGS, BIND_AMBIGUOUS, BIND_QUIT

### CODE REVIEW

File: doc/sphinx_source/modules/internals.rst [hunk 1]
```diff
@@ -125,23 +146,65 @@ Triggering any Bind
...
+| MATCH_PARTIAL     | Check the triggering value against the beginning of the bind mask, ie DIR triggers a mask for DIRECTORY (case insensitive)    |
++-------------------+-------------------------------------------------------------------------------------------------------------------------------+
+| MATCH_EXACT       | Check the triggering value exactly against the bind mask value (case insensitive)                                             |
++-------------------+-------------------------------------------------------------------------------------------------------------------------------+
+| MATCH_CASE        | Check the triggering value exactly against the bind mask value (case sensitive)                                               |
++-------------------+-------------------------------------------------------------------------------------------------------------------------------+
+| MATCH_MASK        | Check if the bind mask is contained within the triggering value, as a wildcarded value                                        |
```
Comment (thommey) [inline doc/sphinx_source/modules/internals.rst]: I don't think "contained" is right, it checks if it matches fully

File: doc/sphinx_source/modules/internals.rst [hunk 2]
```diff
@@ -125,23 +146,65 @@ Triggering any Bind
...
++-------------------+-------------------------------------------------------------------------------------------------------------------------------+
+| MATCH_MODE        | Check if the triggering value is contained within the bind mask, as a wildcarded value                                        |
```
Comment (thommey) [inline doc/sphinx_source/modules/internals.rst]: This mask is like MATCH_MASK until the first space, then it is MATCH_CASE. Specifically written for mode binds where "#channel +o" case is irrelevant for #channel but matters for +o.

File: doc/sphinx_source/modules/internals.rst [hunk 3]
```diff
@@ -125,23 +146,65 @@ Triggering any Bind
...
++-------------------+-------------------------------------------------------------------------------------------------------------------------------+
+| MATCH_CRON        | Check the triggering value against a bind mask formatted as a cron entry, ie "30 7 6 7 * " triggers a mask for "30 7 * * * "  |
```
Comment (thommey) [inline doc/sphinx_source/modules/internals.rst]: The example should not contain * anymore, it is triggered with concrete values like "30 7 6 7 5"

File: doc/sphinx_source/modules/internals.rst [hunk 4]
```diff
@@ -125,23 +146,65 @@ Triggering any Bind
...
+| BIND_STACKABLE    | Allow one mask to be re-used to call multiple Tcl proc. Must be used with HT_STACKABLE_                                       |
++-------------------+-------------------------------------------------------------------------------------------------------------------------------+
+| BIND_WANTRET      | With stacked binds, if the called Tcl proc called returns a '1', halt processing any further binds triggered by the action    |
++-------------------+-------------------------------------------------------------------------------------------------------------------------------+
+| BIND_STACKRET     | Used with BIND_WANTRET; allow stacked binds to continue despite receiving a '1'                                               |
++-------------------+-------------------------------------------------------------------------------------------------------------------------------+
+| BIND_ALTER_ARGS   | Replaces arguments (which ones?) with the result returned from the called Tcl proc                                            |
```
Comment (thommey) [inline doc/sphinx_source/modules/internals.rst]: I can't figure out what this was ever used for and it is not currently used for any bind types in Eggdrop. I propose leaving it undocumented.

File: doc/sphinx_source/modules/internals.rst [hunk 5]
```diff
@@ -125,23 +146,65 @@ Triggering any Bind
...
+| BIND_AMBIGUOUS | The triggering action matched multiple non-stackable binds                                                   |
++----------------+--------------------------------------------------------------------------------------------------------------+
+| BIND_MATCHED   | The bind criteria was met, but the Tcl proc it tried to call could not be found                              |
++----------------+--------------------------------------------------------------------------------------------------------------+
+| BIND_EXECUTED  | The bind criteria was met and the Tcl proc was called                                                        |
++----------------+--------------------------------------------------------------------------------------------------------------+
+| BIND_EXEC_LOG  | The bind criteria was met, the Tcl proc was called, and Eggdrop logged the bind being called                 |
++----------------+--------------------------------------------------------------------------------------------------------------+
+| BIND_QUIT      | The bind was triggered in conjunction with the target leaving the partyline or filesys area (?)              |
```
Comment (thommey) [inline doc/sphinx_source/modules/internals.rst:195]: .quit is bound to a special value CMD_LEAVE that doesn't really trigger a bind but sets the Tcl result to the magic string "break" which is then used to detect this and signal the intention to quit the partyline/filesys by making the bind call return BIND_QUIT

File: doc/sphinx_source/modules/internals.rst (hunk 3 above)
Comment (vanosg) [reply doc/sphinx_source/modules/internals.rst:167]: From the comments in match.c,
```
/* Check if the current time matches a crontab-like specification.
 *
 * mask contains a cron-style series of time fields. The following
 * crontab operators are supported: ranges '-', asterisks '*',
 * lists ',' and steps '/'.
```    
are we sure?

File: doc/sphinx_source/modules/internals.rst (hunk 2 above)
Comment (vanosg) [reply doc/sphinx_source/modules/internals.rst]: Can you write this one up

File: doc/sphinx_source/modules/internals.rst (hunk 5 above)
Comment (vanosg) [reply doc/sphinx_source/modules/internals.rst:195]: Can you write this one up

Comment (thommey) [reply doc/sphinx_source/modules/internals.rst:195]: > Sentinel value to signal `.quit` was triggered to leave the current partyline/filesys area (virtual bind to `CMD_LEAVE`)

File: doc/sphinx_source/modules/internals.rst (hunk 2 above)
Comment (thommey) [reply doc/sphinx_source/modules/internals.rst]: > Special mode for `bind mode` which is like `MATCH_MASK` but case insensitive before the first space ` ` and case sensitive after

**Pull request merged**